homeassistant/components/vicare/types.py

```diff
@@ -64,6 +64,55 @@ def from_ha_preset(
 }
 
 
+class VentilationMode(enum.StrEnum):
+    """ViCare ventilation modes."""
+
+    PERMANENT = "permanent"  # on, speed controlled by program (levelOne-levelFour)
+    VENTILATION = "ventilation"  # activated by schedule
+    SENSOR_DRIVEN = "sensorDriven"  # activated by schedule, override by sensor
+    SENSOR_OVERRIDE = "sensorOverride"  # activated by sensor
+
+    @staticmethod
+    def to_ha_mode(mode: str | None) -> str | None:
+        """Return the mapped Home Assistant mode for the ViCare ventilation mode."""
+        if mode:
+            try:
+                ventilation_mode = VentilationMode(mode)
+            except ValueError:
+                # ignore unsupported / unmapped modes
+                return None
+            return VICARE_TO_HA_MODE_VENTILATION.get(ventilation_mode) if mode else None
+        return None
+
+    @staticmethod
+    def from_ha_mode(ha_mode: str | None) -> str | None:
+        """Return the mapped ViCare ventilation mode for the Home Assistant mode."""
+        for mode in VentilationMode:
+            if VICARE_TO_HA_MODE_VENTILATION.get(VentilationMode(mode)) == ha_mode:
+                return mode
+        return None
+
+
+VICARE_TO_HA_MODE_VENTILATION = {
+    VentilationMode.PERMANENT: "permanent",
+    VentilationMode.VENTILATION: "ventilation",
+    VentilationMode.SENSOR_DRIVEN: "sensor_driven",
+    VentilationMode.SENSOR_OVERRIDE: "sensor_override",
+}
+
```

Comment:
```suggestion
HA_TO_VICARE_MODE_VENTILATION { "permanent": VentilationMode.PERMANENT, "ventilation": VentilationMode.VENTILATION, "sensorDriven": VentilationMode.SENSOR_DRIVEN, "sensorOverride": VentilationMode.SENSOR_OVERRIDE, }
```

Reply:
This is named `HA_TO_VICARE_MODE_VENTILATION` but in fact it's the other way round. `sensorOverride` is the mode coming from vicare.

Reply:
Correct, you can flip the naming of the two dictionaries.

Reply:
Where should this be used?